analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantCast.CSharp8.cs

```diff
@@ -2,15 +2,50 @@
 using System.Collections.Generic;
 using System.Linq;
 
+#nullable enable
 namespace Tests.Diagnostics
 {
     // https://github.com/SonarSource/sonar-dotnet/issues/3273
     public class CastOnNullable
     {
-        public static IEnumerable<string> UsefulCast()
+        public static IEnumerable<string> Array()
         {
             var nullableStrings = new string?[] { "one", "two", null, "three" };
             return nullableStrings.OfType<string>(); // Compliant - filters out the null
```

Comment:
Did you have a look if we can merge our new logic with the one that detects this case?

Reply:
not straight forward

Reply:
Please create a ticket about lighting up the nullability info on ITypeSymbol on the shim layer and link to https://github.com/SonarSource/sonar-dotnet/issues/3273#issuecomment-606693989 and this comment thread.

Reply:
I'm not super clear on what that means and thus don't know what to put in the ticket exactly.

Reply:
https://github.com/SonarSource/sonar-dotnet/issues/6825